File: solutions/java/bowling/1/src/main/java/RollContext.java
public final class RollContext {

    private final Frame frame;
    private final int pins;
    private final int rollIndex;

    public RollContext(final Frame frame, final int pins) {
        this.frame = frame;
        this.pins = pins;
        this.rollIndex = frame.rollCount();
    }

    public Frame getFrame() {
        return this.frame;
    }

    public int getPins() {
        return this.pins;
    }

    public int getRollIndex() {
        return this.rollIndex;
    }
}

File: solutions/ocaml/bowling/1/roll_context.ml
open Base
open Frame

type t = {
  frame : Frame.frame;
  pins : int;
  roll_index : int;
}

let create (frame : Frame.frame) (pins : int) : t =
  { frame; pins; roll_index = Frame.roll_count frame }

let get_frame (context : t) : Frame.frame = context.frame
let get_pins (context : t) : int = context.pins
let get_roll_index (context : t) : int = context.roll_index
